### PR TITLE
[receivers/prometheus] Allow Summary metrics without quantile labels

### PR DESCRIPTION
**Pull request by swiatekm**
**Description:**
Fixing a bug which causes prometheus receiver to completely fail scrapes containing a Summary metric without a quantile label. This isn't required by the Otel schema.

**Link to tracking Issue:** #23995

**Testing:**
Modified existing tests, mostly by changing error conditions.

### CODE REVIEW

File: receiver/prometheusreceiver/internal/util.go [hunk 1]
```diff
@@ -82,7 +81,8 @@ func getBoundary(metricType pmetric.MetricType, labels labels.Labels) (float64,
 	case pmetric.MetricTypeSummary:
 		val = labels.Get(model.QuantileLabel)
 		if val == "" {
-			return 0, errEmptyQuantileLabel
+			// if there's no quantile label, just return 0, which is the default
```
Comment (dashpole) [inline receiver/prometheusreceiver/internal/util.go:84]: Is this needed to be able to handle series without quantile labels?  I thought sum and count never reached this point.

File: receiver/prometheusreceiver/internal/util_test.go [hunk 2]
```diff
@@ -138,23 +138,17 @@ func TestGetBoundary(t *testing.T) {
 			wantValue: 11.71,
 		},
 		{
-			name:    "summary with bucket label",
```
Comment (dashpole) [inline receiver/prometheusreceiver/internal/util_test.go:141]: Why remove these test cases?

File: receiver/prometheusreceiver/internal/util.go (hunk 1 above)
Comment (swiatekm-sumo) [reply receiver/prometheusreceiver/internal/util.go:84]: We call this in `addSeries` here and propagate the error, which causes it for fail the entire transation. An alternative would be to keep this logic as is, but handle the error differently in `addSeries`.

File: receiver/prometheusreceiver/internal/util_test.go (hunk 2 above)
Comment (swiatekm-sumo) [reply receiver/prometheusreceiver/internal/util_test.go:141]: I think their intent was to show that a Summary needed quantile labels (and not bucket labels Histograms have) to be valid. As I'm removing the requirement, the test can just check if no labels at all is fine. The other test is an accidental duplicate I think, it does the exact same thing.

File: receiver/prometheusreceiver/internal/util.go (hunk 1 above)
Comment (dashpole) [reply receiver/prometheusreceiver/internal/util.go:84]: I would love to handle the error differntly in addSeries.  We definitely shouldn't fail the entire scrape if there is a single bad series (assuming that is the current behavior).

File: receiver/prometheusreceiver/internal/util_test.go (hunk 2 above)
Comment (dashpole) [reply receiver/prometheusreceiver/internal/util_test.go:141]: I think we still want to at least log errors if we get a series that isn't sum or count without a quantile label.  I'd like to keep this case if possible.

File: receiver/prometheusreceiver/internal/util.go (hunk 1 above)
Comment (swiatekm-sumo) [reply receiver/prometheusreceiver/internal/util.go:84]: That is indeed the current behaviour. Changing that behaviour is straightforward, but we have a lot of openmetrics validation tests which depend on this behaviour. I'll see if I can find a way to make this a relatively painless change.

Comment (swiatekm-sumo) [reply receiver/prometheusreceiver/internal/util.go:84]: @dashpole I created a separate PR for this change: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24032. Can you have a look?

**Pull request closed without merging**